e2e/dractions/actions.go

```diff
@@ -113,16 +113,36 @@ func DisableProtection(ctx types.Context) error {
 
 func Failover(ctx types.Context) error {
 	d := ctx.Deployer()
-	if d.IsDiscovered() {
-		return FailoverDiscoveredApps(ctx)
-	}
-
 	managementNamespace := ctx.ManagementNamespace()
 	log := ctx.Logger()
+	name := ctx.Name()
+
+	drpcName := name
+	client := util.Ctx.Hub.Client
+	drPolicyName := util.DefaultDRPolicyName
+
+	currentCluster, err := getCurrentCluster(client, managementNamespace, name)
+	if err != nil {
+		return err
+	}
+
+	drpolicy, err := util.GetDRPolicy(client, drPolicyName)
+	if err != nil {
+		return err
+	}
+
+	targetCluster, err := getTargetCluster(client, managementNamespace, drpcName, drpolicy)
+	if err != nil {
+		return err
+	}
 
-	log.Infof("Failing over workload in namespace %q", managementNamespace)
+	log.Infof("Failing over workload from cluster %q to cluster %q", currentCluster, targetCluster)
 
-	return failoverRelocate(ctx, ramen.ActionFailover, ramen.FailedOver)
+	if d.IsDiscovered() {
+		return failoverRelocateDiscoveredApps(ctx, ramen.ActionFailover, ramen.FailedOver, currentCluster, targetCluster)
+	}
```

Comment:
This is worse than the previous code, lets work on this in the next commit. Improving failover and relocate for managed apps is good enough for this commit.